x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/analyses/Regression.java

```diff
@@ -57,63 +56,30 @@ public static Regression fromXContent(XContentParser parser, boolean ignoreUnkno
-                      @Nullable Integer maximumNumberTrees, @Nullable Double featureBagFraction, @Nullable String predictionFieldName,
+    public Regression(String dependentVariable,
+                      BoostedTreeParams boostedTreeParams,
+                      @Nullable String predictionFieldName,
                       @Nullable Double trainingPercent) {
-        this.dependentVariable = Objects.requireNonNull(dependentVariable);
-
-        if (lambda != null && lambda < 0) {
-            throw ExceptionsHelper.badRequestException("[{}] must be a non-negative double", LAMBDA.getPreferredName());
-        }
-        this.lambda = lambda;
-
-        if (gamma != null && gamma < 0) {
-            throw ExceptionsHelper.badRequestException("[{}] must be a non-negative double", GAMMA.getPreferredName());
-        }
-        this.gamma = gamma;
-
-        if (eta != null && (eta < 0.001 || eta > 1)) {
-            throw ExceptionsHelper.badRequestException("[{}] must be a double in [0.001, 1]", ETA.getPreferredName());
-        }
-        this.eta = eta;
-
-        if (maximumNumberTrees != null && (maximumNumberTrees <= 0 || maximumNumberTrees > 2000)) {
-            throw ExceptionsHelper.badRequestException("[{}] must be an integer in [1, 2000]", MAXIMUM_NUMBER_TREES.getPreferredName());
-        }
-        this.maximumNumberTrees = maximumNumberTrees;
-
-        if (featureBagFraction != null && (featureBagFraction <= 0 || featureBagFraction > 1.0)) {
-            throw ExceptionsHelper.badRequestException("[{}] must be a double in (0, 1]", FEATURE_BAG_FRACTION.getPreferredName());
-        }
-        this.featureBagFraction = featureBagFraction;
-
-        this.predictionFieldName = predictionFieldName;
-
         if (trainingPercent != null && (trainingPercent < 1.0 || trainingPercent > 100.0)) {
             throw ExceptionsHelper.badRequestException("[{}] must be a double in [1, 100]", TRAINING_PERCENT.getPreferredName());
         }
+        this.dependentVariable = ExceptionsHelper.requireNonNull(dependentVariable, DEPENDENT_VARIABLE);
+        this.boostedTreeParams = ExceptionsHelper.requireNonNull(boostedTreeParams, BoostedTreeParams.NAME);
+        this.predictionFieldName = predictionFieldName;
         this.trainingPercent = trainingPercent == null ? 100.0 : trainingPercent;
     }
 
     public Regression(String dependentVariable) {
-        this(dependentVariable, null, null, null, null, null, null, null);
+        this(dependentVariable, new BoostedTreeParams(null, null, null, null, null), null, null);
     }
 
     public Regression(StreamInput in) throws IOException {
         dependentVariable = in.readString();
-        lambda = in.readOptionalDouble();
-        gamma = in.readOptionalDouble();
-        eta = in.readOptionalDouble();
-        maximumNumberTrees = in.readOptionalVInt();
-        featureBagFraction = in.readOptionalDouble();
+        boostedTreeParams = new BoostedTreeParams(in);
```

Comment:
We need to add BWC handling here.

Reply:
I think the code (as it is written right now) is backward-compatible as the sequence of StreamInput reads in the old version is the same as in the new version (the new version has the reads wrapped in the `new BoostedTreeParams(in)` constructor.
It would change, however, if I introduced a nested object.

Reply:
Ah, true.